sns/utils.bal

```diff
@@ -0,0 +1,284 @@
+    if (request is http:Request) {
+        http:Response|error httpResponse = amazonSNSClient->post("/", request);
+        xml|error response = handleResponse(httpResponse);
+        if (response is xml) {
+            return response;
+        } else {
+            return response;
+        }
+    } else {
+        return error(REQUEST_ERROR);
+    }
+}
+
+isolated function checkCredentials(string accessKeyId, string secretAccessKey) returns error? {
+    if ((accessKeyId == "") || (secretAccessKey == "")) {
+        return error("Access Key Id or Secret Access Key credential are empty");
+    }
+}
+
+isolated function utcToString(time:Utc utc, string pattern) returns string|error {
+    [int, decimal] [epochSeconds, lastSecondFraction] = utc;
+    int nanoAdjustments = (<int>lastSecondFraction * 1000000000);
+    var instant = ofEpochSecond(epochSeconds, nanoAdjustments);
+    var zoneId = getZoneId(java:fromString("Z"));
+    var zonedDateTime = atZone(instant, zoneId);
+    var dateTimeFormatter = ofPattern(java:fromString(pattern));
+    handle formatString = format(zonedDateTime, dateTimeFormatter);
+    return formatString.toBalString();
+}
+
+# Handles the HTTP response.
+#
+# + httpResponse - Http response or error
+# + return - If successful returns `xml` response. Else returns error
+isolated function handleResponse(http:Response|http:PayloadType|error httpResponse) returns @untainted xml|error {
+    if (httpResponse is http:Response) {
+        if (httpResponse.statusCode == http:STATUS_NO_CONTENT) {
+            return error ResponseHandleFailed(NO_CONTENT_SET_WITH_RESPONSE_MSG);
+        }
+        var xmlResponse = httpResponse.getXmlPayload();
+        if (xmlResponse is xml) {
+            if (httpResponse.statusCode == http:STATUS_OK) {
+                return xmlResponse;
+            } else {
+                xmlns "http://sns.amazonaws.com/doc/2010-03-31/" as ns;
+                string xmlResponseErrorCode = httpResponse.statusCode.toString();
+                string responseErrorMessage = (xmlResponse/<ns:'error>/<ns:message>/*).toString();
+                string errorMsg = "status code" + ":" + xmlResponseErrorCode + 
+                    ";" + " " + "message" + ":" + " " + 
+                    responseErrorMessage;
+                return error(errorMsg);
+            }
+        } else {
+            return error(RESPONSE_PAYLOAD_IS_NOT_XML_MSG);
+        }
+    } else if (httpResponse is http:PayloadType) {
+        return error(UNREACHABLE_STATE);
+    } else {
+        return error(ERROR_OCCURRED_WHILE_INVOKING_REST_API_MSG, httpResponse);
+    }
+}
+
+# Set topic attributes to a map of string to add as query parameters.
+#
+# + parameters - Parameter map
+# + attributes - TopicAttributes to convert to a map of string
+# + return - If successful returns `map<string>` response. Else returns error
+isolated function setTopicAttributes(map<string> parameters, TopicAttributes attributes) returns map<string> {
```

Comment:
There can be a way to avoid different mentods of setting attributes. The body of those functions seems to be same. Only the type of the attributes input is different.